firedrake/ufl_expr.py

```diff
@@ -139,7 +139,8 @@ def derivative(form, u, du=None, coefficient_derivatives=None):
     """
     # TODO: What about Constant?
     u_is_x = isinstance(u, ufl.SpatialCoordinate)
-    if not u_is_x and len(u.split()) > 1 and set(extract_coefficients(form)) & set(u.split()):
+    uc, = extract_coefficients(u)
```

Comment:
This extraction fails for coordinate derivatives `if u_is_x`, can you handle that?